rocAL/rocAL/include/reader.h

```diff
@@ -23,61 +23,93 @@ THE SOFTWARE.
+    void set_start_end_frame_vector(std::vector<std::tuple<unsigned, unsigned>> start_end_frame) { _start_end_frame_vector = start_end_frame; }
     size_t get_shard_count() { return _shard_count; }
     size_t get_shard_id() { return _shard_id; }
     size_t get_batch_size() { return _batch_count; }
+    size_t get_sequence_length() { return _sequence_length; }
+    size_t get_frame_step() { return _video_frame_step; }
+    size_t get_frame_stride() { return _video_frame_stride; }
+    size_t get_video_count() { return _video_count; }
+    size_t get_video_frame_rate() { return _frame_rate; }
+    std::vector<size_t> get_video_frames_count() { return _frame_count; }
+    size_t get_total_frames_count() { return _total_frames_count; }
+    std::vector<std::string> get_video_file_names() { return _video_file_names; }
+    std::vector<std::tuple<unsigned, unsigned>> get_start_end_frame_vector() { return _start_end_frame_vector; }
     std::string path() { return _path; }
-    std::string json_path() { return _json_path;}
-    std::map<std::string, std::string> feature_key_map() {return _feature_key_map; }
-    void set_file_prefix(const std::string &prefix) {_file_prefix = prefix;}
-    std::string file_prefix() {return _file_prefix;}
-    std::shared_ptr<MetaDataReader> meta_data_reader() {return _meta_data_reader;}
+    std::string json_path() { return _json_path; }
+    std::map<std::string, std::string> feature_key_map() { return _feature_key_map; }
+    void set_file_prefix(const std::string &prefix) { _file_prefix = prefix; }
+    std::string file_prefix() { return _file_prefix; }
+    std::shared_ptr<MetaDataReader> meta_data_reader() { return _meta_data_reader; }
 private:
     StorageType _type = StorageType::FILE_SYSTEM;
     std::string _path = "";
     std::string _json_path = "";
     std::map<std::string, std::string> _feature_key_map;
-    size_t _shard_count= 1 ;
+    size_t _shard_count = 1;
     size_t _shard_id = 0;
-    size_t _batch_count = 1;//!< The reader will repeat images if necessary to be able to have images in multiples of the _batch_count.
+    size_t _batch_count = 1;     //!< The reader will repeat images if necessary to be able to have images in multiples of the _batch_count.
+    size_t _sequence_length = 1; // Video reader module sequence length
+    size_t _video_frame_step;
+    size_t _video_frame_stride = 1;
+    unsigned _video_count;
+    std::vector<size_t> _frame_count;
+    size_t _frame_rate;
+    size_t _total_frames_count;
+    std::vector<std::string> _video_file_names;
+    std::vector<std::tuple<unsigned, unsigned>> _start_end_frame_vector;
     bool _shuffle = false;
     bool _loop = false;
     std::string _file_prefix = ""; //!< to read only files with prefix. supported only for cifar10_data_reader and tf_record_reader
     std::shared_ptr<MetaDataReader> _meta_data_reader = nullptr;
 };
 
-class Reader {
+class Reader
```

Comment:
better to have a separate reader class for video

Reply:
We will add this in next PR which will have separate Video Reader class.